server/middleware/setUpCookies.ts

```diff
@@ -3,11 +3,18 @@ import { Router } from 'express'
 
 export default function setUpCookies(): Router {
   const router = Router()
+  const cookiePreferencesPostRoute = '/cookie-preferences'
 
   router.use(cookieParser())
 
   router.use((req, res, next) => {
+    if (req.url !== cookiePreferencesPostRoute) {
+      req.session.returnTo = req.originalUrl
+    }
     res.locals.acceptAnalyticsCookies = req.cookies?.acceptAnalyticsCookies
+    res.locals.analyticsCookiesPreference = req.cookies?.acceptAnalyticsCookies === 'true' ? 'accepted' : 'rejected'
```

Comment:
I wonder if this reads right: "My preference is accepted/rejected". Might also be worth making it clear that this local variable is for string interpolation - not just relabelling `acceptAnalyticsCookies` - e.g. `analyticsCookiesDecisionText` (or even `analyticsCookiesPreferenceText`, because the `Text` makes it feel less like the grammar need to work as a sentence)

Reply:
Also, given this intercepts every request, I guess setting it to `"rejected"` if the user hasn't set a preference yet is a bit confusing here? Is it worth putting this in a conditional?